rcsb/workflow/bcif/task_functions.py

```diff
@@ -0,0 +1,414 @@
+            os.chmod(bcifFilePath, 0o777)
+        except PermissionError as e:
+            pass
+        counter[0] += 1
+    except Exception as e:
+        logger.exception(str(e))
+    finally:
+        # remove cif file
+        if remotePath.startswith("http") and os.path.exists(cifFilePath):
+            os.unlink(cifFilePath)
+        # remove temp files
+        if counter[0] >= maxTempFiles:
+            removeTempFiles(tempPath=dtemp)
+            counter[0] = 0
+
+
+def getBcifFilePath(
+    pdbId, outfileSuffix, updateBase, contentType, outputContentType, outputHash
+):
+    bcifFileName = "%s%s" % (pdbId, outfileSuffix)
+    bcifFilePath = None
+    if contentType == "pdb":
+        if outputContentType and outputHash:
+            bcifFilePath = os.path.join(
+                updateBase, contentType, pdbId[-3:-1], bcifFileName
+            )
+        elif outputContentType:
+            bcifFilePath = os.path.join(updateBase, contentType, bcifFileName)
+        elif outputHash:
+            bcifFilePath = os.path.join(updateBase, pdbId[-3:-1], bcifFileName)
+        else:
+            bcifFilePath = os.path.join(updateBase, bcifFileName)
+    elif contentType == "csm":
+        if outputContentType and outputHash:
+            bcifFilePath = os.path.join(
+                updateBase,
+                contentType,
+                pdbId[0:2],
+                pdbId[-6:-4],
+                pdbId[-4:-2],
+                bcifFileName,
+            )
+        elif outputContentType:
+            bcifFilePath = os.path.join(
+                updateBase,
+                contentType,
+                bcifFileName,
+            )
+        elif outputHash:
+            bcifFilePath = os.path.join(
+                updateBase,
+                pdbId[0:2],
+                pdbId[-6:-4],
+                pdbId[-4:-2],
+                bcifFileName,
+            )
+        else:
+            bcifFilePath = os.path.join(
+                updateBase,
+                bcifFileName,
+            )
+    return bcifFilePath
+
+
+def convert(
+    infile: str, outfile: str, workpath: str, dictionaryApi: DictionaryApi
+) -> bool:
+    mu = MarshalUtil(workPath=workpath)
+    data = mu.doImport(infile, fmt="mmcif")
+    try:
+        result = mu.doExport(outfile, data, fmt="bcif", dictionaryApi=dictionaryApi)
+        if not result:
+            raise Exception()
+    except Exception as e:
+        raise Exception("error during bcif conversion: %s" % str(e)) from e
+    return True
+
+
+def deconvert(
+    infile: str, outfile: str, workpath: str, dictionaryApi: DictionaryApi
+) -> bool:
+    mu = MarshalUtil(workPath=workpath)
+    data = mu.doImport(infile, fmt="bcif")
+    try:
+        result = mu.doExport(outfile, data, fmt="mmcif", dictionaryApi=dictionaryApi)
+        if not result:
+            raise Exception()
+    except Exception as e:
+        raise Exception("error during bcif conversion: %s" % str(e)) from e
```

Comment:
This try/catch doesn't have much effect (just changing the message slightly). I'd rather remove it. It will improve readability.